divide_and_conquer/binary_exponentiation.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file
+ * @brief Implementation of Binary Exponentiation
+ * @details
+ * Binary Exponentiation is a way to calculate exponential
+ * calculations using much less time complexity (O(logn))
+ * using a divide and conquer technique through recursion.
+ * [https://cp-algorithms.com/algebra/binary-exp.html]
+ * @author [Timothy Chiles](https://github.com/TSChiles)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+
+/**
+ * @namespace divide_and_conquer
+ */
+namespace divide_and_conquer {
+
+/**
+ * @brief Function documentation
+ * @param param1 is the base number
+ * @param param2 is the exponent
+ * @returns the base to the exponent operation
+ */
+    long long expo(long long param1, long long param2) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂